add_nyt_bestseller_tag.py

```diff
@@ -20,87 +20,107 @@
-            self.logger.info('found edition {} for isbn {}'.format(
-                repr(found_edition),
-                book_isbn ))
-            if not found_edition:
-                self.logger.info('Not found edition for ISBN {}'.format(book_isbn))
-                return False
-            self.logger.info('Found edition for ISBN {}'.format(book_isbn))
-            return True
-        except Exception as e:
-            # TODO check the exception type
-            job.logger.exception('There is no book with ISBN {}'.format(book_isbn))
-            return False
+            work.subjects.append(new_tag)
+            self.logger.info(
+                'Successfully appended new NYT tag to work {}'.format(
+                    work.olid))
+        except AttributeError:
+            work.subjects = [new_tag]
+            self.logger.exception(
+                'Failed to append subjects list for work {} but '
+                'SUCCESSFULLY CREATED new subjects list with NYT tag '.format(
+                    work.olid))
 
     def request_book_import_by_isbn(self, book_isbn) -> None:
         """ Makes request to the book_isbn
         https://openlibrary.org/isbn/{book_isbn} """
         url = 'https://openlibrary.org/isbn/{}'.format(book_isbn)
         try:
             requests.get(url)
-            job.logger.info('Made request to {}'.format(url))
+            self.logger.info('Made request to {}'.format(url))
         except Exception as e:
             self.logger.error('Failed to make request to {}'.format(url))
 
+    def save_job_resutls(self, job_results):
+        self.logger.info('Job execution results: {}'.format(repr(job_results)))
+        with open('add_nyt_bestseller_result.json', 'w', encoding='utf-8') as f:
+            json.dump(job_results, f, ensure_ascii=False, indent=4)
+
     def run(self) -> None:  # overwrites the AbstractBotJob run method
         self.dry_run_declaration()
-
+        job_results = {'input_file': self.args.file, 'books_imported': 0,
+                       'tags_added': 0, 'tags_already_exist': 0}
         comment = 'Add NYT bestseller tag'
         with open(self.args.file, 'r') as fin:
             bestsellers_data = json.load(fin)
-            if 'error' not in bestsellers_data:
-                for bestseller_group_record in bestsellers_data:
-                    new_tag = 'nyt:{}={}'.format(
-                        bestseller_group_record['list_name_encoded'],
-                        bestseller_group_record['published_date'] )
-                    for bstslr_record_isbn in bestseller_group_record['isbns']:
-                        if self.book_exists_in_ol(bstslr_record_isbn):
+            for bestseller_group_record in bestsellers_data:
+                new_tag = 'nyt:{}={}'.format(
+                    bestseller_group_record['list_name_encoded'],
+                    bestseller_group_record['published_date'])
+                for bstslr_record_isbn in bestseller_group_record['isbns']:
```

Comment:
Could we pull out most of the code under the isbn into a separate function or two so that it's not so deeply nested with indents?

Reply:
Makes sense, will do!

Reply:
Done